core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcSchema.java

```diff
@@ -433,6 +433,12 @@ private static RelDataType sqlType(RelDataTypeFactory typeFactory, int dataType,
                 typeFactory.createSqlType(SqlTypeName.ANY), true);
       }
       return typeFactory.createArrayType(component, -1);
+    case OTHER:
```

Comment:
On most databases the type string will be upper-case, "GEOMETRY".

Reply:
Yes, the use of OTHER with a "geometry" type name seems specific to the postgresql jdbc driver and postgis. I guess they did this to distinguish postgis geometry types from the regular geometry types implemented in postgresql. https://www.postgresql.org/docs/current/datatype-geometric.html

I will clarify this with a comment and try to investigate regular geometry data types a bit more.